src/dotnet/APIView/APIView/CodeFileRenderer.cs

```diff
@@ -22,27 +22,48 @@ private void Render(List<CodeLine> list, IEnumerable<CodeFileToken> node)
         {
             var stringBuilder = new StringBuilder();
             string currentId = null;
+            bool isDocumentation = false;
+            bool isDeprecatedLine = false;
+
             foreach (var token in node)
             {
-                if (token.Kind == CodeFileTokenKind.Newline)
-                {
-                    list.Add(new CodeLine(stringBuilder.ToString(), currentId));
-                    currentId = null;
-                    stringBuilder.Clear();
-                }
-                else
+                switch(token.Kind)
                 {
-                    if (token.DefinitionId != null)
-                    {
-                        currentId = token.DefinitionId;
-                    }
+                    // One assumption here is that documention line will not have actual code in same line
+                    // Assumption is based on that requirement to hide documention on demand
+                    case CodeFileTokenKind.Newline:
+                    case CodeFileTokenKind.DocumentRangeEnd:
+                        list.Add(new CodeLine(stringBuilder.ToString(), currentId, isDocumentation));
+                        currentId = null;
+                        stringBuilder.Clear();
+                        if (token.Kind == CodeFileTokenKind.DocumentRangeEnd)
+                            isDocumentation = false;
+                        break;
+
+                    case CodeFileTokenKind.DocumentRangeStart:
+                        isDocumentation = true;
+                        break;
+
+                    case CodeFileTokenKind.DeprecatedRangeStart:
+                        isDeprecatedLine = true;
```

Comment:
We shouldn't mark the entire line deprecated, only the tokens inside DeprecatedRangeStart-DeprecatedRangeEnd

Reply:
Actually this variable name is confusing. But implementation only marks tokens inside the range as deprecated. I will update the name to avoid confusion.